sw/airborne/modules/glide_wing_lock/glide_wing_lock.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) Kevin van Hecke
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file "modules/glide_wing_lock/glide_wing_lock.c"
+ * @author Kevin van Hecke
+ * Locks the wing of an ornicopter so it can glide.
+ */
+
+#include "modules/glide_wing_lock/glide_wing_lock.h"
+
+#include "pprzlink/messages.h"
+#include "mcu_periph/uart.h"
+#include "subsystems/datalink/downlink.h"
+
+#include "mcu_periph/adc.h"
+#include "subsystems/commands.h"
+#include "subsystems/radio_control.h"
+#include "autopilot.h"
+
+struct adc_buf adcbuf;
+
+#ifndef WING_POS_DOWN_THRESH
+#define WING_POS_DOWN_THRESH 100
+#endif
+#ifndef WING_POS_LOCK_MIN_THRESH
+#define WING_POS_LOCK_MIN_THRESH 2000
+#endif
+#ifndef WING_POS_LOCK_MAX_THRESH
+#define WING_POS_LOCK_MAX_THRESH 2100
+#endif
+#ifndef WING_POS_LOCK_SWITCH
+#define WING_POS_LOCK_SWITCH RADIO_AUX2
+#endif
+
+void glide_wing_lock_init(void) {
+  adc_buf_channel(ADC_CHANNEL_MOTORSENSOR, &adcbuf, 1);
+}
+
+void glide_wing_lock_event() {    
+  static int lockstate = 0;
+  if (radio_control.values[RADIO_AUX2] > (MIN_PPRZ / 2)) { // check glide switch
```

Comment:
I suggest to make this configurable if someone doesn't have an AUX2 channel on his radio